test/test_array_string_long.cpp

```diff
@@ -108,6 +108,14 @@ TEST_TYPES(ArrayStringLong_Basic, non_nullable, nullable)
     CHECK_EQUAL("", c.get(4));
     CHECK_EQUAL("", c.get(5));
 
+    c.add("hi");
+    c.add("hello");
+
+    // Test that you can set a value obtained from the same blob
+    c.set(0, c.get(6));
```

Comment:
What happens if you use this new syntax, but you address a source which is in a *different* blob? This should be documented somewhere.

It's a nice speed optimization, but I think this set method that allows this should be renamed into something else, because it hints users that it's allowed on strings, datetime, etc, etc, too.

Btw. we could add new asserts to set_string() to guard against such mistakes (I've seen people make it a few times) :) (in a different PR, maybe I'll just make an issue)